traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -0,0 +1,247 @@
+		active    => $params->{active},
+		parent_id => $params->{parentId}
+	};
+
+	my $rs = $tenant->update($values);
+	if ($rs) {
+		my $response;
+		$response->{id}          = $rs->id;
+		$response->{name}        = $rs->name;
+		$response->{active}      = \$rs->active;
+		$response->{parentId}    = $rs->parent_id;
+		$response->{lastUpdated} = $rs->last_updated;
+		&log( $self, "Updated Tenant name '" . $rs->name . "' for id: " . $rs->id, "APICHANGE" );
+		return $self->success( $response, "Tenant update was successful." );
+	}
+	else {
+		return $self->alert("Tenant update failed.");
+	}
+
+}
+
+
+sub create {
+	my $self   = shift;
+	my $params = $self->req->json;
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	my $name = $params->{name};
+	if ( !defined($name) ) {
+		return $self->alert("Tenant name is required.");
+	}
+
+	my $parent_id = $params->{parentId};
+	if ( !defined($parent_id) ) {
+		return $self->alert("Parent Id is required.");
+	}
+
+	my $existing = $self->db->resultset('Tenant')->search( { name => $name } )->get_column('name')->single();
+	if ($existing) {
+		return $self->alert("A tenant with name \"$name\" already exists.");
+	}
+
+	my $is_active = exists($params->{active})? $params->{active} : 0; #optional, if not set use default
+	
+	if ( !$is_active && !defined($parent_id)) {
+		return $self->alert("Root user cannot be in-active.");
+	}
+	
+	my $values = {
+		name 		=> $params->{name} ,
+		active		=> $is_active,
+		parent_id 	=> $params->{parentId}
+	};
+
+	my $insert = $self->db->resultset('Tenant')->create($values);
+	my $rs = $insert->insert();
+	if ($rs) {
+		my $response;
+		$response->{id}          	= $rs->id;
+		$response->{name}        	= $rs->name;
+		$response->{active}        	= \$rs->active;
+		$response->{parentId}           = $rs->parent_id;
+		$response->{lastUpdated} 	= $rs->last_updated;
+
+		&log( $self, "Created Tenant name '" . $rs->name . "' for id: " . $rs->id, "APICHANGE" );
+
+		return $self->success( $response, "Tenant create was successful." );
+	}
+	else {
+		return $self->alert("Tenant create failed.");
+	}
+
+}
+
+
+sub delete {
+	my $self = shift;
+	my $id     = $self->param('id');
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	my $tenant = $self->db->resultset('Tenant')->find( { id => $id } );
+	if ( !defined($tenant) ) {
+		return $self->not_found();
+	}	
+	my $name = $self->db->resultset('Tenant')->search( { id => $id } )->get_column('name')->single();
+	
+	my $existing_child = $self->db->resultset('Tenant')->search( { parent_id => $id } )->get_column('name')->first();
+	if ($existing_child) {
+		return $self->alert("Tenant '$name' has children tenant(s): e.g '$existing_child'. Please update these tenants and retry.");
+	}
+
+	#The order of the below tests is intentional - allowing UT to cover all cases - TODO(nirs) remove this comment when a full "tenancy" UT is added, including permissions and such (no use in putting effort into it yet)
```

Comment:
No Problem. 
Note that you'll need to be careful on the merge when the next 3 PRs are added - as the order of the tests is important for the UT, I hope to fix it later on.